### PR TITLE
Fix: Multisig single chain view

### LINKED ISSUE #2550: Flexible multisig. Rework wallet details for multisig
Foe multisigs with single account wallet details should look different https://www.figma.com/design/HgFuj8QZd0xqEbkG10DIfs/Flexible-multisig-%7C-Milestone-4?node-id=842-38475&t=kjeUlo152YAs4Ksj-4

### PR DESCRIPTION
**Pull request by tuul-wq**
- Updated Multisig wallet details for a single chain
- Better TS support for `dictionary` utility function
- Correct determination of signatories (wallets, contacts, others)

closes #2550 

image

### CODE REVIEW

File: src/renderer/features/wallets/WalletSelect/ui/WalletGroup.tsx [hunk 1]
```diff
@@ -54,8 +54,9 @@ export const WalletGroup = ({ type, wallets }: Props) => {
                   )
                 }
                 onClick={() => walletSelectModel.events.walletSelected(wallet.id)}
-                onInfoClick={() => walletSelectModel.events.walletIdSet(wallet.id)}
```
Comment (tuul-wq) [inline src/renderer/features/wallets/WalletSelect/ui/WalletGroup.tsx]: Now we can pass whatever button is required as children and not rely on tricky behavior of `WalletCardMd`

File: src/renderer/shared/lib/utils/__tests__/arrays.test.ts [hunk 2]
```diff
@@ -23,126 +23,204 @@ describe('shared/lib/onChainUtils/arrays', () => {
     expect(array1).toEqual([100]);
     expect(array2).toEqual([100]);
   });
+});
+
+describe('addUniq', () => {
```
Comment (tuul-wq) [inline src/renderer/shared/lib/utils/__tests__/arrays.test.ts:28]: Separated `describe` groups and provided tests for `dictionary`

File: src/renderer/shared/ui-entities/AccountExplorer/AccountExplorers.tsx [hunk 3]
```diff
@@ -2,20 +2,23 @@ import { type PropsWithChildren, memo } from 'react';
...
-  chain: Chain;
+  chain?: Chain;
 }>;
 
 export const AccountExplorers = memo(({ accountId, chain, children }: Props) => {
   const { t } = useI18n();
-  const { explorers } = chain;
-  const address = toAddress(accountId, { prefix: chain.addressPrefix });
+
+  const explorers = chain?.explorers || RootExplorers;
```
Comment (tuul-wq) [inline src/renderer/shared/ui-entities/AccountExplorer/AccountExplorers.tsx]: `AccountExplorers` should use RootExplorers if chain is absent (any root account relies on this)

File: src/renderer/widgets/WalletDetails/model/wallet-provider-model.ts [hunk 4]
```diff
@@ -59,62 +59,44 @@ const $vaultAccounts = combine(
   { skipVoid: false },
 );
 
-const $signatoryContacts = combine(
+const $signatories = combine(
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/model/wallet-provider-model.ts]: Just form 3 groups of signatories that eliminates any duplication (previously duplicates existed)

File: src/renderer/shared/ui-entities/AccountExplorer/AccountExplorers.tsx (hunk 3 above)
Comment (johnthecat) [reply src/renderer/shared/ui-entities/AccountExplorer/AccountExplorers.tsx]: Why? No, it shouldn't

File: src/renderer/shared/ui-entities/AccountExplorer/AccountExplorers.tsx [hunk 5]
```diff
@@ -2,20 +2,23 @@ import { type PropsWithChildren, memo } from 'react';
...
+  const addressPrefix = chain?.addressPrefix || SS58_DEFAULT_PREFIX;
```
Comment (johnthecat) [inline src/renderer/shared/ui-entities/AccountExplorer/AccountExplorers.tsx]: Let's discuss this code

**Review by johnthecat — CHANGES_REQUESTED** [review]

**Pull request merged**